components/ws-proxy/pkg/proxy/workspacerouter.go

```diff
@@ -136,15 +156,38 @@ func matchBlobserveHostHeader(wsHostSuffix string, headerProvider hostHeaderProv
 		}
 
 		matches := r.FindStringSubmatch(hostname)
-		return len(matches) >= 1
+		if len(matches) < 1 {
+			return false
+		}
+		matches = pathPortRegex.FindStringSubmatch(req.URL.Path)
+		if len(matches) < 3 {
+			matches = pathDebugRegex.FindStringSubmatch(req.URL.Path)
+		}
+		if len(matches) < 3 {
+			return true
+		}
+
+		if m.Vars == nil {
+			m.Vars = make(map[string]string)
+		}
+		m.Vars[workspacePathPrefixIdentifier] = strings.TrimRight(matches[0], "/")
+		m.Vars[workspaceIDIdentifier] = matches[2]
+		if matches[1] == "debug-" {
+			m.Vars[debugWorkspaceIdentifier] = "true"
+		} else {
+			m.Vars[workspacePortIdentifier] = matches[1]
+		}
+
+		return true
```

Comment:
Could we add a unit test to cover this part?

Reply:
@jenting there are already unit tests they are still passing, i also added a new test for a new case, see changes to `components/ws-proxy/pkg/proxy/workspacerouter_test.go` and `components/ws-proxy/pkg/proxy/routes_test.go`